### PR TITLE
Job splitters should retain original geometries

### PR DESCRIPTION
**Pull request by VincentVerelst**
@kvantricht, updated both `split_job_hex` and `split_job_s2grid` so that they now:
- Retain the CRS of the original geodataframe
- Retain the geometries of the original geodataframe
- Use Web Mercator instead of lat lon to calculate the centroids, so that the warnings are gone

Also added unit tests for both.

### CODE REVIEW

File: src/openeo_gfmap/manager/job_splitters.py [hunk 1]
```diff
@@ -60,17 +60,23 @@ def split_job_s2grid(
...
+    original_crs = polygons.crs
+
+    # Transform to web mercator, to calculate the centroid
+    polygons = polygons.to_crs(epsg=3857)
+
+    polygons["centroid"] = polygons.geometry.centroid
 
     # Dataset containing all the S2 tiles, find the nearest S2 tile for each point
     s2_grid = load_s2_grid()
+    s2_grid = s2_grid.to_crs(epsg=3857)
```
Comment (kvantricht) [inline src/openeo_gfmap/manager/job_splitters.py:72]: is this not a slow step for the entire s2 grid?

**Review by kvantricht — APPROVED** [review]
Just one remark on the speed of reprojecting s2 grid.

File: src/openeo_gfmap/manager/job_splitters.py (hunk 1 above)
Comment (GriffinBabe) [reply src/openeo_gfmap/manager/job_splitters.py:72]: @VincentVerelst Maybe if this is a speed issue you could upload a second s2 grid in the artifactory with the CRS reprojected, and then change the `load_s2_grid()` function in gfmap to accept a parameter `web_mercator: bool = False`

Comment (kvantricht) [reply src/openeo_gfmap/manager/job_splitters.py:72]: May be worth the effort indeed.

Comment (VincentVerelst) [reply src/openeo_gfmap/manager/job_splitters.py:72]: Thanks for the suggestion, I'll already update that.

Do have to note that the largest bottleneck is typically not the CRS conversion of the s2 grid, but rather the CRS conversion of the dataframe to split itself.

For example, for `2018_BEL_LPIS-Flanders_POLY_110` the old job_splitter takes 10s to run, while the new one takes 20s.

**Pull request merged**